src/main/java/com/n1analytics/paillier/EncryptedNumber.java

```diff
@@ -416,4 +416,28 @@ public boolean equals(EncryptedNumber o) {
             context.equals(o.context) &&
             ciphertext.equals(o.ciphertext));
   }
+
+  /**
+   * Return a byte array representing the underlying raw number.
+   *
+   * WARNING: This method may return unencrypted information! Under no
+   * circumstances should the return value be sent anywhere.
+   *
+   * WARNING: This method is not intended for general use and may be
+   * removed in a future release. Think twice before using it.
+   */
+  public byte[] toByteArrayUnsafe() {
+    return ciphertext.toByteArray();
+  }
+
+  /**
+   * Return true or false according to whether this EncryptedNumber is
+   * safe for transmission.
+   *
+   * WARNING: This method is not intended for general use and may be
+   * removed in a future release. Think twice before using it.
+   */
+  public boolean getIsSafe() {
```

Comment:
Thanks for the clarification. What I meant by 'internal' process is that you are doing all this to improve the performance of the library itself. You are not wanting these additional APIs to enhance the usage of the library. In other words, what you need should ideally not even be made aware to a 'normal' user. 

Therefore, my concerns go even further. You mention that you primarily deal with arrays of encrypted numbers, so I assume that you may serialise the entire vector into a single byte array which you then hand down to your GPU library. If you do this, you may actually create a a few intermittent byte arrays and a few additional copies between all them. In addition, I wonder how much extra 'massaging' you need to do in your GPU library to properly interpret the big integer serialisation the Java library performs in it's `toByteArray`. 

In other words, this entire process seems to be rather inefficient and kind of counter-productive to the overall goal you have - make this bird fly faster. And given all the effort you already put into the GPU library, I wonder if we shouldn't spend a bit more effort on the passing back and forth across the JNI boundary.

Now it's perfectly possible, that I completely misjudging that overhead